goflags.go

```diff
@@ -163,21 +210,52 @@ func (flagSet *FlagSet) generateDefaultConfig() []byte {
 	return bytes.TrimSuffix(configBuffer.Bytes(), []byte("\n\n"))
 }
 
-// readConfigFile reads the config file and returns any flags
-// that might have been set by the config file.
-//
-// Command line flags however always take precedence over config file ones.
-func (flagSet *FlagSet) readConfigFile(filePath string) error {
+// validate config file for missing property or type mis-match
+func (flagSet *FlagSet) validateDefaultConfig(filePath string) error {
+	config, err := unMarshalDefaultConfig(filePath)
+	if err != nil {
+		return err
+	}
+	for k, v := range config {
+		flagVal, ok := flagSet.flagKeys.values[k]
+		if !ok {
+			var ErrProperty = errors.New("property:%s not found in the config file")
+			return errors.New(fmt.Sprintf(ErrProperty.Error(), k))
```

Comment:
Since it requires formatting, we can use:
```go
fmt.Errorf("property:%s not found in the config file", k)
```

Reply:
Done